arangod/Transaction/Methods.cpp

```diff
@@ -1466,10 +1466,23 @@ OperationResult transaction::Methods::insertLocal(
     ManagedDocumentResult result;
     TRI_voc_tick_t resultMarkerTick = 0;
     TRI_voc_rid_t revisionId = 0;
+    auto needsLock = [this,&collection]() -> bool {return !isLocked(collection, AccessMode::Type::WRITE);};
+
+    Result res = collection->insert( this, value, result, options
+                                   , resultMarkerTick, needsLock(), revisionId
+                                   );
 
-    Result res =
-        collection->insert(this, value, result, options, resultMarkerTick,
-                           !isLocked(collection, AccessMode::Type::WRITE), revisionId);
+    if(options.overwrite && res.is(TRI_ERROR_ARANGO_UNIQUE_CONSTRAINT_VIOLATED)){
+      // RepSert Case - unique_constraint violated -> maxTick has not changed -> try replace
+      resultMarkerTick = 0;
+      TRI_voc_rid_t revision;
+      ManagedDocumentResult previousDocumentResult; // return OLD/NEW?
+      _localHints.unset(transaction::Hints::Hint::SINGLE_OPERATION);
```

Comment:
the "unset" should not be necessary here, and it indicates something is wrong at the call site.
Any caller should not have set the "SINGLE_OPERATION" hint in case an overwrite operation is attempted.
We should even add an assert that the "SINGLE_OPERATION" hint is not set when an overwrite is attempted.

Reply:
Removing the hint just here may have bad side effects, which we would like to avoid if possible.